src/client/datascience/ipywidgets/commonMessageCoordinator.ts

```diff
@@ -127,6 +130,22 @@ export class CommonMessageCoordinator {
 
     private sendLoadFailureTelemetry(payload: ILoadIPyWidgetClassFailureAction) {
         try {
+            let errorMessage: string = payload.error.toString();
+            if (!payload.isOnline) {
+                errorMessage = localize.DataScience.loadClassFailedWithNoInternet().format(
+                    payload.moduleName,
+                    payload.moduleVersion
+                );
+            } else if (!payload.cdnsUsed) {
+                errorMessage = localize.DataScience.enableCDNForWidgetsSetting().format(
+                    payload.moduleName,
+                    payload.moduleVersion
+                );
+            }
+            traceError(`Widget load failure ${errorMessage}`, payload);
+
+            this.appSell.showErrorMessage(errorMessage).then(noop, noop);
```

Comment:
Seems weird to put up a message in sendLoadFailureTelemetry. Unexpected. Would it be better to split out the error message into a separate function?

Reply:
Renamed the function